File: app/src/main/java/com/github/weatherapp/common/util/CollectionUtils.java
package com.github.weatherapp.common.util;

public class CollectionUtils {

    public static <T> int indexOf(T[] array, T item) {
        int i = 0;

        for (T it : array) {
            if (it.equals(item)) return i;
            ++i;
        }

        return -1;
    }
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/util/CollectionUtils.kt
package com.github.weatherapp.util

object CollectionUtils {

    fun <T> indexOf(array: Array<T>, item: T): Int {
        var i = 0

        for (it in array) {
            if (it == item) return i
            ++i
        }

        return -1
    }
}
